geps/gep-1911.md

```diff
@@ -0,0 +1,260 @@
+
+Implementations MUST recognize support for multiple protocols on the same port.
+
+Currently Kubernetes `Service` API does not allow different `appProtocol` values for the same port. At this time there seems to be interest in changing `appProtocol` to be a list in order to faciliate this use-case.
+
+### Default Protocols
+
+If a backend protocol isn't specified an implementation MAY infer the backend protocol through its own means. 
+
+### Unsupported/Incompatible Protocols
+
+If a Route attached to a Gateway is not able to sending traffic to the backend using the specified protocol then it must set it's RouteConditionType `Accepted` to `False`, with a RouteConditionReason `UnsupportedProtocol`
+
+### Supporting new protocols
+
+To add support for a new protocol it should first become a Kubernetes Standard Application Protocol by updating the [KEP-3726](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3726-standard-application-protocols). If the suggested protocol is not suited to have a `kubernetes.io/*` prefix then the Gateway API MAY support the new protocol using it's own prefix `gateway-api.kubernetes.io/*`.  Please make a PR to this GEP.
+
+## Alternatives
+
+### Single Meta-resource 
+
+The first pass of this GEP proposed a new meta-resource [GEP-713](geps/gep-713.md) called `BackendProtocol`.
+
+This allows end-users to specify a list of ports and a list of corresponding protocols that that single
+port supports.
+
+This was dropped in favour of supporting Kubernetes Standard Application Protocols. The Go and YAML API are
+presented below for posterity.
+
+#### Go API
+
+```go
+type BackendProtocol struct {
+  // Standard type metadata
+  metav1.TypeMeta `json:",inline"`
+
+  // Standard object's metadata
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  // Spec defines the behaviour of this meta-resource
+  Spec BackendProtocolSpec `json:"spec,omitempty"`
+}
+
+type BackendProtocolSpec struct {
+  // Specifies the API resource who's protocols are being
+  // explicitly defined
+  TargetRef LocalObjectReference `json:"targetRef"`
+
+  // Ports contains a list of protocols for a specific ports
+  // When multiple protocols reference the same port the 
+  // entries earlier in the list should have higher priority
+  //
+  // ie. attempt to use h2c otherwise fallback on http1
+  //  ports: 
+  //   - port: 80
+  //     http2: 
+  //       encrypted: false # enable h2c
+  //   - port: 80
+  //     http1: {}          # We also listen to http1 on port 80
+  //
+  Ports []BackendPort `json:"ports,omitempty"`
+}
+
+type BackendPort struct {
+  // Number or name of the port who's protocol is being defined
+  Port intstr.IntOrString
+
+  PortProtocol `json:",inline"`
+}
+
+type PortProtocol struct {
+  HTTP2     *HTTP2PortProtocol    `json:"http2"`
+  HTTP1     *HTTP1PortProtocol    `json:"http1"`
+  GRPC      *GRPCPortProtocol     `json:"grpc"`
```

Comment:
Is it encrypted ? How is it different from h2c from gateway perspective ?

Reply:
you're reviewing an alternative proposal - this isn't what's being recommended anymore